src/melee.cpp

```diff
@@ -1532,51 +1532,95 @@ item &Character::best_shield()
     }
     blocks_left--;
 
-    martial_arts_data.ma_ongethit_effects(
-        *this ); // fire martial arts on-getting-hit-triggered effects
+    // fire martial arts on-getting-hit-triggered effects
     // these fire even if the attack is blocked (you still got hit)
+    martial_arts_data.ma_ongethit_effects( *this );
 
     // This bonus absorbs damage from incoming attacks before they land,
     // but it still counts as a block even if it absorbs all the damage.
     float total_phys_block = mabuff_block_bonus();
+
     // Extract this to make it easier to implement shields/multiwield later
     item &shield = best_shield();
+    block_bonus = blocking_ability( shield );
     bool conductive_shield = shield.conductive();
-    bool unarmed = shield.has_flag( "UNARMED_WEAPON" );
+    bool unarmed = weapon.has_flag( "UNARMED_WEAPON" );
+    bool force_unarmed = martial_arts_data.is_force_unarmed();
+
+    int melee_skill = get_skill_level( skill_melee );
+    int unarmed_skill = get_skill_level( skill_unarmed );
+
+    // Check if we are going to block with an item. This could
+    // be worn equipment with the BLOCK_WHILE_WORN flag.
+    const bool item_blocking = !shield.is_null();
 
-    // This gets us a number between:
-    // str ~0 + skill 0 = 0
-    // str ~20 + skill 10 + 10(unarmed skill or weapon bonus) = 40
     int block_score = 1;
-    // Remember if we're using a weapon or a limb to block.
-    // So that we don't suddenly switch that for any reason.
-    const bool weapon_blocking = !shield.is_null();
-    if( !martial_arts_data.is_force_unarmed() && weapon_blocking ) {
-        /** @EFFECT_STR increases attack blocking effectiveness with a weapon */
-
-        /** @EFFECT_MELEE increases attack blocking effectiveness with a weapon */
-        block_bonus = blocking_ability( shield );
-        block_score = str_cur + block_bonus + get_skill_level( skill_melee );
-    } else if( martial_arts_data.can_limb_block( *this ) ) {
-        /** @EFFECT_STR increases attack blocking effectiveness with a limb */
 
-        /** @EFFECT_UNARMED increases attack blocking effectiveness with a limb */
-        block_score = str_cur + get_skill_level( skill_melee ) + get_skill_level( skill_unarmed );
+    /** @EFFECT_STR increases attack blocking effectiveness with a limb or worn/wielded item */
+    /** @EFFECT_UNARMED increases attack blocking effectiveness with a limb or worn/wielded item */
+    if( ( unarmed || force_unarmed ) ) {
+        if( martial_arts_data.can_limb_block( *this ) ) {
+            // block_bonus for limb blocks will be added when the limb is decided
+            block_score = str_cur + melee_skill + unarmed_skill;
+        } else if( item_blocking ) {
+            // We can still block with a worn item while unarmed. Use higher of melee and unarmed
+            block_score = str_cur + block_bonus + ( melee_skill > unarmed_skill ?
+                                                    melee_skill :  unarmed_skill );
```

Comment:
```suggestion
            block_score = str_cur + block_bonus + std::max( melee_skill, unarmed_skill );
```

Please don't use trinaries to implement min/max functions.